smtp/init.lua

```diff
@@ -152,6 +152,29 @@ local function addr_spec(mailbox)
     return mailbox:match('^.*[ \t]*<(.-)>[ \t]*$') or mailbox
 end
 
+-- According to RFC 822#3.3 subject must contain characters with codes 0-127:
+-- https://tools.ietf.org/html/rfc822
+-- For compatibility with legacy mail relays simply base64 encode subject field
+-- if char codes >127 is present
+local function encode_subject(subj)
+    local is_gt_127 = function(text)
+        for idx = 1, #text do
+            if text:byte(idx) > 127 then
+               return true
+            end
+        end
+        return false
+    end
+
+    if subj and type(subj) == 'string' and subj~='' and is_gt_127(subj) then
+        local encoded = digest.base64_encode(subj)
+        if encoded then
```

Comment:
base64_encode() raises an error in case of an error. As I see, it never returns `nil`.

Reply:
Removed the condition.